JDK27 throws a java.lang.NoClassDefFoundError under -Xjit:count=200,optLevel=scorching, but runs normally in default

## Environment:
- java -Xshareclasses:none -version
openjdk version "27-internal" 2026-09-15
OpenJDK Runtime Environment (fastdebug build 27-internal-adhoc.sjz.jdk27u-j9)
Eclipse OpenJ9 VM (fastdebug build 27-internal-adhoc.sjz.jdk27u-j9-master-f5a91e34d2, JRE 27 Linux amd64-64-Bit Compressed References 20260119_000000 (JIT enabled, AOT enabled)
OpenJ9   - f5a91e34d2
OMR      - ccb25976d
JCL      - 67c6256a511 based on jdk-27+5)

- OS: Linux 5.8.0-63-generic, x86_64
- CPU: amd64, 48 logical CPUs
- Build GCC: 9.4
## Key :
```
- Options:  -Xjit:count=0,optLevel=scorching  &  Default
```
## Steps to reproduce
Run with OpenJ9 fastdebug build:
   - diff run 1:
    ```
     jdk/bin/java -Xjit:count=200,optLevel=scorching  -Xshareclasses:none  Test
    ```
   - diff run2 :
    ```
     jdk/bin/java  -Xshareclasses:none  Test
    ```
## Result
With   -Xjit:count=200,optLevel=scorching  Test, the result is:
```
Exception in thread "main" java.lang.NoClassDefFoundError: com.ibm.oti.util.Msg (initialization failure)
        at java.base/java.lang.J9VMInternals.initializationAlreadyFailed(J9VMInternals.java:161)
        at java.base/java.lang.J9VMInternals.cloneThrowable(J9VMInternals.java:241)
        at java.base/java.lang.J9VMInternals.copyThrowable(J9VMInternals.java:274)
        at java.base/java.lang.J9VMInternals.copyThrowable(J9VMInternals.java:256)
        at java.base/java.lang.J9VMInternals.recordInitializationFailure(J9VMInternals.java:192)
        at Test.m1(Test.java:144)
        at Test.m3(Test.java:226)

```
With Test, the result is:
```
normally
```
## Source code
```
import jdk.incubator.vector.*;
import java.util.Arrays;

public class Test {
    static class A {
        void m() {
        }
    }
    static class B extends Test.A {
        void m() {
        }
    }
    static void m1(boolean deopt, Test.A a) {
        ShortVector lv_ShortVector_1768938424674_101485693542737491 = null;
        ShortVector lv_ShortVector_1768938424674_101485693537278330 = null;
        Integer lv_Integer_1768938424675_101485693549319230 = Integer.valueOf(437);
        char lv_char_1768938424668_101485693484733700 = (char)63;
        short[] lv_short_arr_1768938424674_101485693542882550 = new short[AllFuzzerDefs_1768938423442_98100.ARRAY_SIZE];
        String lv_String_1768938424673_101485693531214930 = new String("asdfasf123123");
        float lv_float_1768938424672_101485693522955620 = 521.02484f;
        float lv_float_1768938424673_101485693528406531 = 521.02484f;
        int lv_int_1768938424671_101485693509545663 = 19;
        int lv_int_1768938424670_101485693502795762 = 19;
        int lv_int_1768938424658_101485693376469460 = 19;
        int lv_int_1768938424665_101485693451334221 = 19;
        char[] lv_char_arr_1768938424670_101485693503726030 = new char[AllFuzzerDefs_1768938423442_98100.ARRAY_SIZE];
        for (int TmpVar_10148569365893209_1768938424686 = 0; TmpVar_10148569365893209_1768938424686 < lv_short_arr_1768938424674_101485693542882550.length; TmpVar_10148569365893209_1768938424686++) {
            lv_short_arr_1768938424674_101485693542882550[TmpVar_10148569365893209_1768938424686] = ((short) ((((short) (TmpVar_10148569365893209_1768938424686)) * 10) + 34));
        }
        for (int TmpVar_10148569366173407_1768938424686 = 0; TmpVar_10148569366173407_1768938424686 < lv_char_arr_1768938424670_101485693503726030.length; TmpVar_10148569366173407_1768938424686++) {
            lv_char_arr_1768938424670_101485693503726030[TmpVar_10148569366173407_1768938424686] = ((char) ((((char) (TmpVar_10148569366173407_1768938424686)) * 5) + 12));
        }
        long l0, l1, l2, l3, l4, l5, l6, l7, l8, l9, l10, l11, l12,
        l13, l14, l15, l16, l17, l18, l19, l20, l21, l22, l23, l24,
        l25, l26, l27, l28, l29, l30, l31, l32, l33, l34, l35, l36,
        l37, l38, l39, l40, l41, l42, l43, l44, l45, l46, l47, l48,
        l49, l50, l51, l52, l53, l54, l55, l56, l57, l58, l59, l60,
        l61, l62, l63, l64, l65, l66, l67, l68, l69, l70, l71, l72,
        l73, l74, l75, l76, l77, l78, l79, l80, l81, l82, l83, l84,
        l85, l86, l87, l88, l89, l90, l91, l92, l93, l94, l95, l96,
        l97, l98, l99, l100, l101, l102, l103, l104, l105, l106, l107,
        l108, l109, l110, l111, l112, l113, l114, l115, l116, l117,
        l118, l119, l120, l121, l122, l123, l124, l125, l126, l127,
        l128, l129, l130, l131, l132, l133, l134, l135, l136, l137,
        l138, l139, l140, l141, l142, l143, l144, l145, l146, l147,
        l148, l149, l150, l151, l152, l153, l154, l155, l156, l157,
        l158, l159, l160, l161, l162, l163, l164, l165, l166, l167,
        l168, l169, l170, l171, l172, l173, l174, l175, l176, l177,
        l178, l179, l180, l181, l182, l183, l184, l185, l186, l187,
        l188, l189, l190, l191, l192, l193, l194, l195, l196, l197,
        l198, l199, l200, l201, l202, l203, l204, l205, l206, l207,
        l208, l209, l210, l211, l212, l213, l214, l215, l216, l217,
        l218, l219, l220, l221, l222, l223, l224, l225, l226, l227,
        l228, l229, l230, l231, l232, l233, l234, l235, l236, l237,
        l238, l239, l240, l241, l242, l243, l244, l245, l246, l247,
        l248, l249, l250, l251, l252, l253, l254, l255, l256, l257,
        l258, l259, l260, l261, l262, l263, l264, l265, l266, l267,
        l268, l269, l270, l271, l272, l273, l274, l275, l276, l277,
        l278, l279, l280, l281, l282, l283, l284, l285, l286, l287,
        l288, l289, l290, l291, l292, l293, l294, l295, l296, l297,
        l298, l299, l300, l301, l302, l303, l304, l305, l306, l307,
        l308, l309, l310, l311, l312, l313, l314, l315, l316, l317,
        l318, l319, l320, l321, l322, l323, l324, l325, l326, l327,
        l328, l329, l330, l331, l332, l333, l334, l335, l336, l337,
        l338, l339, l340, l341, l342, l343, l344, l345, l346, l347,
        l348, l349, l350, l351, l352, l353, l354, l355, l356, l357,
        l358, l359, l360, l361, l362, l363, l364, l365, l366, l367,
        l368, l369, l370, l371, l372, l373, l374, l375, l376, l377,
        l378, l379, l380, l381, l382, l383, l384, l385, l386, l387,
        l388, l389, l390, l391, l392, l393, l394, l395, l396, l397,
        l398, l399, l400, l401, l402, l403, l404, l405, l406, l407,
        l408, l409, l410, l411, l412, l413, l414, l415, l416, l417,
        l418, l419, l420, l421, l422, l423, l424, l425, l426, l427,
        l428, l429, l430, l431, l432, l433, l434, l435, l436, l437,
        l438, l439, l440, l441, l442, l443, l444, l445, l446, l447,
        l448, l449, l450, l451, l452, l453, l454, l455, l456, l457,
        l458, l459, l460, l461, l462, l463, l464, l465, l466, l467,
        l468, l469, l470, l471, l472, l473, l474, l475, l476, l477,
        l478, l479, l480, l481, l482, l483, l484, l485, l486, l487,
        l488, l489, l490, l491, l492, l493, l494, l495, l496, l497,
        l498, l499, l500, l501, l502, l503, l504, l505, l506, l507,
        l508, l509, l510, l511;

        long ll0, ll1, ll2, ll3, ll4, ll5, ll6, ll7, ll8, ll9, ll10, ll11, ll12,
        ll13, ll14, ll15, ll16, ll17, ll18, ll19, ll20, ll21, ll22, ll23, ll24,
        ll25, ll26, ll27, ll28, ll29, ll30, ll31, ll32, ll33, ll34, ll35, ll36,
        ll37, ll38, ll39, ll40, ll41, ll42, ll43, ll44, ll45, ll46, ll47, ll48,
        ll49, ll50, ll51, ll52, ll53, ll54, ll55, ll56, ll57, ll58, ll59, ll60,
        ll61, ll62, ll63, ll64, ll65, ll66, ll67, ll68, ll69, ll70, ll71, ll72,
        ll73, ll74, ll75, ll76, ll77, ll78, ll79, ll80, ll81, ll82, ll83, ll84,
        ll85, ll86, ll87, ll88, ll89, ll90, ll91, ll92, ll93, ll94, ll95, ll96,
        ll97, ll98, ll99, ll100, ll101, ll102, ll103, ll104, ll105, ll106, ll107,
        ll108, ll109, ll110, ll111, ll112, ll113, ll114, ll115, ll116, ll117,
        ll118, ll119, ll120, ll121, ll122, ll123, ll124, ll125, ll126, ll127,
        ll128, ll129, ll130, ll131, ll132, ll133, ll134, ll135, ll136, ll137,
        ll138, ll139, ll140, ll141, ll142, ll143, ll144, ll145, ll146, ll147,
        ll148, ll149, ll150, ll151, ll152, ll153, ll154, ll155, ll156, ll157,
        ll158, ll159, ll160, ll161, ll162, ll163, ll164, ll165, ll166, ll167,
        ll168, ll169, ll170, ll171, ll172, ll173, ll174, ll175, ll176, ll177,
        ll178, ll179, ll180, ll181, ll182, ll183, ll184, ll185, ll186, ll187,
        ll188, ll189, ll190, ll191, ll192, ll193, ll194, ll195, ll196, ll197,
        ll198, ll199, ll200, ll201, ll202, ll203, ll204, ll205, ll206, ll207,
        ll208, ll209, ll210, ll211, ll212, ll213, ll214, ll215, ll216, ll217,
        ll218, ll219, ll220, ll221, ll222, ll223, ll224, ll225, ll226, ll227,
        ll228, ll229, ll230, ll231, ll232, ll233, ll234, ll235, ll236, ll237,
        ll238, ll239, ll240, ll241, ll242, ll243, ll244, ll245, ll246, ll247,
        ll248, ll249, ll250, ll251, ll252, ll253, ll254, ll255, ll256, ll257,
        ll258, ll259, ll260, ll261, ll262, ll263, ll264, ll265, ll266, ll267,
        ll268, ll269, ll270, ll271, ll272, ll273, ll274, ll275, ll276, ll277,
        ll278, ll279, ll280, ll281, ll282, ll283, ll284, ll285, ll286, ll287,
        ll288, ll289, ll290, ll291, ll292, ll293, ll294, ll295, ll296, ll297,
        ll298, ll299, ll300, ll301, ll302, ll303, ll304, ll305, ll306, ll307,
        ll308, ll309, ll310, ll311, ll312, ll313, ll314, ll315, ll316, ll317,
        ll318, ll319, ll320, ll321, ll322, ll323, ll324, ll325, ll326, ll327,
        ll328, ll329, ll330, ll331, ll332, ll333, ll334, ll335, ll336, ll337,
        ll338, ll339, ll340, ll341, ll342, ll343, ll344, ll345, ll346, ll347,
        ll348, ll349, ll350, ll351, ll352, ll353, ll354, ll355, ll356, ll357,
        ll358, ll359, ll360, ll361, ll362, ll363, ll364, ll365, ll366, ll367,
        ll368, ll369, ll370, ll371, ll372, ll373, ll374, ll375, ll376, ll377,
        ll378, ll379, ll380, ll381, ll382, ll383, ll384, ll385, ll386, ll387,
        ll388, ll389, ll390, ll391, ll392, ll393, ll394, ll395, ll396, ll397,
        ll398, ll399, ll400, ll401, ll402, ll403, ll404, ll405, ll406, ll407,
        ll408, ll409, ll410, ll411, ll412, ll413, ll414, ll415, ll416, ll417,
        ll418, ll419, ll420, ll421, ll422, ll423, ll424, ll425, ll426, ll427,
        ll428, ll429, ll430, ll431, ll432, ll433, ll434, ll435, ll436, ll437,
        ll438, ll439, ll440, ll441, ll442, ll443, ll444, ll445, ll446, ll447,
        ll448, ll449, ll450, ll451, ll452, ll453, ll454, ll455, ll456, ll457,
        ll458, ll459, ll460, ll461, ll462, ll463, ll464, ll465, ll466, ll467,
        ll468, ll469, ll470, ll471, ll472, ll473, ll474, ll475, ll476, ll477,
        ll478, ll479, ll480, ll481, ll482, ll483, ll484, ll485, ll486, ll487,
        ll488, ll489, ll490, ll491, ll492, ll493, ll494, ll495, ll496, ll497,
        ll498, ll499, ll500, ll501, ll502, ll503, ll504, ll505, ll506, ll507,
        ll508, ll509, ll510, ll511;
        a.m();
        if (deopt) {
            Test.do_load = true;
            while (!Test.load_done) {
                for (lv_int_1768938424658_101485693376469460 = 1; lv_int_1768938424658_101485693376469460 < 333; lv_int_1768938424658_101485693376469460++) {

                    for (lv_int_1768938424670_101485693502795762 = 0; lv_int_1768938424670_101485693502795762 < 333; lv_int_1768938424670_101485693502795762++) {
                        
                        lv_ShortVector_1768938424674_101485693537278330 = ShortVector.fromArray(ShortVector.SPECIES_PREFERRED, lv_short_arr_1768938424674_101485693542882550, 5);
                        
                    }
                }
            } 
        }
        System.out.println("Integer lv_Integer_1768938424675_101485693549319230:: " + String.valueOf(lv_Integer_1768938424675_101485693549319230));
        System.out.println("char lv_char_1768938424668_101485693484733700:: " + ((int) (lv_char_1768938424668_101485693484733700)));
        for (int TmpVar_10148569366019628_1768938424686 = 0; TmpVar_10148569366019628_1768938424686 < lv_short_arr_1768938424674_101485693542882550.length; TmpVar_10148569366019628_1768938424686 = (2 + (TmpVar_10148569366019628_1768938424686 * 3)) / 2) {
            System.out.println((("short[] lv_short_arr_1768938424674_101485693542882550:: at " + TmpVar_10148569366019628_1768938424686) + " ") + lv_short_arr_1768938424674_101485693542882550[TmpVar_10148569366019628_1768938424686]);
        }
        System.out.println(("String lv_String_1768938424673_101485693531214930:: `" + lv_String_1768938424673_101485693531214930) + "`");
        System.out.println("float lv_float_1768938424672_101485693522955620:: " + lv_float_1768938424672_101485693522955620);
        System.out.println("float lv_float_1768938424673_101485693528406531:: " + lv_float_1768938424673_101485693528406531);
        System.out.println("int lv_int_1768938424671_101485693509545663:: " + lv_int_1768938424671_101485693509545663);
        System.out.println("int lv_int_1768938424670_101485693502795762:: " + lv_int_1768938424670_101485693502795762);
        System.out.println("int lv_int_1768938424658_101485693376469460:: " + lv_int_1768938424658_101485693376469460);
        System.out.println("int lv_int_1768938424665_101485693451334221:: " + lv_int_1768938424665_101485693451334221);
        for (int TmpVar_10148569366207740_1768938424686 = 0; TmpVar_10148569366207740_1768938424686 < lv_char_arr_1768938424670_101485693503726030.length; TmpVar_10148569366207740_1768938424686 = (2 + (TmpVar_10148569366207740_1768938424686 * 3)) / 2) {
            System.out.println((("char[] lv_char_arr_1768938424670_101485693503726030:: at " + TmpVar_10148569366207740_1768938424686) + " ") + ((int) (lv_char_arr_1768938424670_101485693503726030[TmpVar_10148569366207740_1768938424686])));
        }
    }
    static void m2(boolean deopt, Test.A a) {
        long l0, l1, l2, l3, l4, l5, l6, l7, l8, l9, l10, l11, l12,
        l13, l14, l15, l16, l17, l18, l19, l20, l21, l22, l23, l24,
        l25, l26, l27, l28, l29, l30, l31, l32, l33, l34, l35, l36,
        l37, l38, l39, l40, l41, l42, l43, l44, l45, l46, l47, l48,
        l49, l50, l51, l52, l53, l54, l55, l56, l57, l58, l59, l60,
        l61, l62, l63, l64, l65, l66, l67, l68, l69, l70, l71, l72,
        l73, l74, l75, l76, l77, l78, l79, l80, l81, l82, l83, l84,
        l85, l86, l87, l88, l89, l90, l91, l92, l93, l94, l95, l96,
        l97, l98, l99, l100, l101, l102, l103, l104, l105, l106, l107,
        l108, l109, l110, l111, l112, l113, l114, l115, l116, l117,
        l118, l119, l120, l121, l122, l123, l124, l125, l126, l127,
        l128, l129, l130, l131, l132, l133, l134, l135, l136, l137,
        l138, l139, l140, l141, l142, l143, l144, l145, l146, l147,
        l148, l149, l150, l151, l152, l153, l154, l155, l156, l157,
        l158, l159, l160, l161, l162, l163, l164, l165, l166, l167,
        l168, l169, l170, l171, l172, l173, l174, l175, l176, l177,
        l178, l179, l180, l181, l182, l183, l184, l185, l186, l187,
        l188, l189, l190, l191, l192, l193, l194, l195, l196, l197,
        l198, l199, l200, l201, l202, l203, l204, l205, l206, l207,
        l208, l209, l210, l211, l212, l213, l214, l215, l216, l217,
        l218, l219, l220, l221, l222, l223, l224, l225, l226, l227,
        l228, l229, l230, l231, l232, l233, l234, l235, l236, l237,
        l238, l239, l240, l241, l242, l243, l244, l245, l246, l247,
        l248, l249, l250, l251, l252, l253, l254, l255, l256, l257,
        l258, l259, l260, l261, l262, l263, l264, l265, l266, l267,
        l268, l269, l270, l271, l272, l273, l274, l275, l276, l277,
        l278, l279, l280, l281, l282, l283, l284, l285, l286, l287,
        l288, l289, l290, l291, l292, l293, l294, l295, l296, l297,
        l298, l299, l300, l301, l302, l303, l304, l305, l306, l307,
        l308, l309, l310, l311, l312, l313, l314, l315, l316, l317,
        l318, l319, l320, l321, l322, l323, l324, l325, l326, l327,
        l328, l329, l330, l331, l332, l333, l334, l335, l336, l337,
        l338, l339, l340, l341, l342, l343, l344, l345, l346, l347,
        l348, l349, l350, l351, l352, l353, l354, l355, l356, l357,
        l358, l359, l360, l361, l362, l363, l364, l365, l366, l367,
        l368, l369, l370, l371, l372, l373, l374, l375, l376, l377,
        l378, l379, l380, l381, l382, l383, l384, l385, l386, l387,
        l388, l389, l390, l391, l392, l393, l394, l395, l396, l397,
        l398, l399, l400, l401, l402, l403, l404, l405, l406, l407,
        l408, l409, l410, l411, l412, l413, l414, l415, l416, l417,
        l418, l419, l420, l421, l422, l423, l424, l425, l426, l427,
        l428, l429, l430, l431, l432, l433, l434, l435, l436, l437,
        l438, l439, l440, l441, l442, l443, l444, l445, l446, l447,
        l448, l449, l450, l451, l452, l453, l454, l455, l456, l457,
        l458, l459, l460, l461, l462, l463, l464, l465, l466, l467,
        l468, l469, l470, l471, l472, l473, l474, l475, l476, l477,
        l478, l479, l480, l481, l482, l483, l484, l485, l486, l487,
        l488, l489, l490, l491, l492, l493, l494, l495, l496, l497,
        l498, l499, l500, l501, l502, l503, l504, l505, l506, l507,
        l508, l509, l510, l511;
        try {
            Test.m2(deopt, a);
        } catch (java.lang.StackOverflowError e) {
            Test.m3(deopt, a);
        }
    }
    static void m3(boolean deopt, Test.A a) {
        
        a.m();
        Test.m1(deopt, a);
    static volatile boolean thread_started = false;
    static volatile boolean do_load = false;
    static volatile boolean load_done = false;
   
    public static void main(java.lang.String[] args) {
        java.lang.Thread thread = new java.lang.Thread() {
            public void run() {
                Test.thread_started = true;
                while (!Test.do_load) {
                } 
                new Test.B();
                Test.load_done = true;
            }
        };
        thread.start();
        while (!Test.thread_started) {
        } 
        Test.A a = new Test.A();
        for (int i = 0; i < 5000; i++) {
            Test.m3(false, a);
            Test.m1(false, a);
        }
        Test.m2(true, a);
        java.lang.System.out.println("TEST PASSED");
    }
}

class AllFuzzerDefs_1768938423442_98100 {

    public static int ARRAY_SIZE = 80;


}


```

I tried doing this by calling `preInitializeMsg() `inside `completeInitialization()`. In this approach, `preInitializeMsg() `is invoked after the class loader and thread initialization are completed. The method forces `com.ibm.oti.util.Msg` to load and execute its static initializer early using `Class.forName()`.
```
--- a/jcl/src/java.base/share/classes/java/lang/J9VMInternals.java
+++ b/jcl/src/java.base/share/classes/java/lang/J9VMInternals.java
@@ -107,6 +107,8 @@ final class J9VMInternals {
                ClassLoader.completeInitialization();
                threadCompleteInitialization();
 
+               preInitializeMsg();
+
                /*[IF JAVA_SPEC_VERSION >= 9]*/
                System.initGPUAssist();
 
@@ -181,6 +183,14 @@ final class J9VMInternals {
                /*[ENDIF] (11 <= JAVA_SPEC_VERSION) & (JAVA_SPEC_VERSION <= 17) */
        }
 
+       private static void preInitializeMsg() {
+               try {
+                       ClassLoader loader = J9VMInternals.class.getClassLoader();
+                       Class.forName("com.ibm.oti.util.Msg", true, loader);
+               } catch (Throwable t) {
+               }
+       }
+

``` 
error:
```
Entering m2, deopt=true
Entering m2, deopt=true
Caught StackOverflowError in m2
Entering m3, deopt=true
Inside A.m()
Entering m1, deopt=true
Initializing arrays...
Inside A.m()
Deopt enabled, entering load loop...
Thread finished setup
Exception in thread "main" java.lang.ExceptionInInitializerError
	at java.base/jdk.internal.foreign.Utils.<clinit>(Utils.java:84)
	at java.base/jdk.internal.foreign.layout.AbstractLayout.requirePowerOfTwoAndGreaterOrEqualToOne(AbstractLayout.java:146)
	at java.base/jdk.internal.foreign.layout.AbstractLayout.<init>(AbstractLayout.java:58)
	at java.base/jdk.internal.foreign.layout.ValueLayouts$AbstractValueLayout.<init>(ValueLayouts.java:81)
	at java.base/jdk.internal.foreign.layout.ValueLayouts$OfAddressImpl.<init>(ValueLayouts.java:317)
	at java.base/jdk.internal.foreign.layout.ValueLayouts$OfAddressImpl.of(ValueLayouts.java:356)
	at java.base/java.lang.foreign.ValueLayout.<clinit>(ValueLayout.java:416)
	at jdk.incubator.vector/jdk.incubator.vector.ShortVector.<clinit>(ShortVector.java:60)
	at Test.m1(Test.java:147)
	at Test.m3(Test.java:232)
	at Test.m2(Test.java:226)
	at Test.m2(Test.java:223)
	at Test.m2(Test.java:223)
...
	at Test.m2(Test.java:223)
	at Test.m2(Test.java:223)
	at Test.m2(Test.java:223)
	at Test.main(Test.java:259)
Caused by: java.lang.StackOverflowError
	at java.base/jdk.internal.util.ReferencedKeyMap.removeStaleReferences(ReferencedKeyMap.java:332)
	at java.base/jdk.internal.util.ReferencedKeyMap.get(ReferencedKeyMap.java:201)
	at java.base/jdk.internal.util.ReferencedKeySet.get(ReferencedKeySet.java:164)
	at java.base/java.lang.invoke.MethodType.makeImpl(MethodType.java:394)
	at java.base/java.lang.invoke.MethodType.insertParameterTypes(MethodType.java:502)
	at java.base/java.lang.invoke.MethodType.appendParameterTypes(MethodType.java:515)
	at java.base/java.lang.invoke.DirectMethodHandle.makePreparedLambdaForm(DirectMethodHandle.java:264)
	at java.base/java.lang.invoke.DirectMethodHandle.preparedLambdaForm(DirectMethodHandle.java:232)
	at java.base/java.lang.invoke.DirectMethodHandle.preparedLambdaForm(DirectMethodHandle.java:212)
	at java.base/java.lang.invoke.DirectMethodHandle.preparedLambdaForm(DirectMethodHandle.java:226)
	at java.base/java.lang.invoke.DirectMethodHandle.make(DirectMethodHandle.java:109)
	at java.base/java.lang.invoke.MethodHandles$Lookup.getDirectMethodCommon(MethodHandles.java:3889)
	at java.base/java.lang.invoke.MethodHandles$Lookup.getDirectMethod(MethodHandles.java:3844)
	at java.base/java.lang.invoke.MethodHandles$Lookup.findStatic(MethodHandles.java:2529)
	at java.base/jdk.internal.foreign.Utils.<clinit>(Utils.java:77)
	... 137 more
```

--- a/jcl/src/java.base/share/classes/java/lang/J9VMInternals.java
+++ b/jcl/src/java.base/share/classes/java/lang/J9VMInternals.java
@@ -320,6 +320,10 @@ final class J9VMInternals {
 	 * @return a copy of the Throwable
 	 */
 	private static Throwable copyThrowable(Throwable throwable) {
+		/* Avoid copying StackOverflowError to prevent additional stack use during an already exhausted stack condition. */
+		if (throwable instanceof StackOverflowError) {
+			return throwable;
+		}
 		HashMap hashMapThrowable = new HashMap();
 		/*[PR CMVC 199629] Exception During Class Initialization Not Handled Correctly */
 		return copyThrowable(throwable, hashMapThrowable);
@@ -335,6 +339,10 @@ final class J9VMInternals {
 	 * @return a copy of the Throwable or itself if it has been cloned already
 	 */
 	private static Throwable copyThrowable(Throwable throwable, HashMap hashMapThrowable) {
+		/* Avoid copying StackOverflowError to prevent additional stack use during an already exhausted stack condition. */
+		if (throwable instanceof StackOverflowError) {
+			return throwable;
+		}
 		if (hashMapThrowable.get(throwable) != null) {
 			//	stop recursive call here when the throwable has been cloned
 			return	throwable;

